### PR TITLE
Add Crate name to Rust main breakpoint

### PR DESCRIPTION
**Pull request by justdak**
Please #221 for more details.

Signed-off-by: Dmitry Kudryavtsev <dmitry.kudryavtsev@arsysop.ru>

### CODE REVIEW

File: org.eclipse.corrosion/src/org/eclipse/corrosion/ui/InputComponent.java [hunk 1]
```diff
@@ -104,7 +106,10 @@ public void createProjectSelection() {
 			Object[] results = dialog.getResult();
 			if (returnCode == 0 && results.length > 0) {
 				text.setText(((IProject) results[0]).getName());
-				editListener.modifyText(null);
+				Event event = new Event();
```
Comment (mickaelistria) [inline org.eclipse.corrosion/src/org/eclipse/corrosion/ui/InputComponent.java]: We should avoid manually creating events.

File: org.eclipse.corrosion/src/org/eclipse/corrosion/debug/RustDebugTabGroup.java [hunk 2]
```diff
@@ -29,17 +29,36 @@
...
+		@Override
+		public void projectChanged(String projectName) {
+			if (false == projectName.equalsIgnoreCase(".cargo")) { //$NON-NLS-1$
+				fStopInMainSymbol.setText(projectName + "::main"); //$NON-NLS-1$
+			}
+		}
+
+	}
+
+	protected interface ProjectInputListener {
```
Comment (mickaelistria) [inline org.eclipse.corrosion/src/org/eclipse/corrosion/debug/RustDebugTabGroup.java]: Can we avoid adding a new listener class here if we have only one internal use-case for it? It seems to me like over-engineering.

File: org.eclipse.corrosion/src/org/eclipse/corrosion/debug/RustDebugTabGroup.java [hunk 3]
```diff
@@ -29,17 +29,36 @@
...
 		@Override
 		public void setDefaults(ILaunchConfigurationWorkingCopy config) {
 			super.setDefaults(config);
 			config.setAttribute(ICDTLaunchConfigurationConstants.ATTR_DEBUGGER_STOP_AT_MAIN, false);
 			config.setAttribute(IGDBLaunchConfigurationConstants.ATTR_DEBUG_NAME, "rust-gdb"); //$NON-NLS-1$
 		}
+
+		@Override
+		public void projectChanged(String projectName) {
+			if (false == projectName.equalsIgnoreCase(".cargo")) { //$NON-NLS-1$
```
Comment (mickaelistria) [inline org.eclipse.corrosion/src/org/eclipse/corrosion/debug/RustDebugTabGroup.java]: Shouldn't the condition be something like `if (project.getFile("Cargo.toml").exists()) {...}` ?

**Comment by mickaelistria** [comment]
> Do you think that this is an unrealistic use-case?

I don't know. I think if it's both your first idea and mine, it sounds like a not-too-bad one. So we can adopt it and adapt later if some users complain.

**Comment by justdak** [comment]
Ok, I will try to re-work this PR to translate `main` to `project::main` with RustDebugDelegate.
I still have one question.  CDT check the **"Stop on startup at:"** checkbox by default. Is there any reason why this checkbox is unchecked by default in Corrosion?

**Comment by norru** [comment]
> this checkbox is unchecked by default in Corrosion?

I *always* unckeck this box. I have rarely found a single use case for this being checked.

File: org.eclipse.corrosion/src/org/eclipse/corrosion/debug/RustDebugDelegate.java [hunk 4]
```diff
@@ -159,6 +159,12 @@
 		wc.setAttribute(ICDTLaunchConfigurationConstants.ATTR_PROGRAM_NAME, executable.getAbsolutePath());
 		wc.setAttribute(ICDTLaunchConfigurationConstants.ATTR_WORKING_DIRECTORY, workingDirectory.getAbsolutePath());
 		wc.setAttribute(ICDTLaunchConfigurationConstants.ATTR_LOCATION, project.getLocation().toString());
+		
+		String stopInMainSymbol = configuration.getAttribute(ICDTLaunchConfigurationConstants.ATTR_DEBUGGER_STOP_AT_MAIN_SYMBOL, ""); //$NON-NLS-1$
+		if (false == projectName.equals(".cargo") && (stopInMainSymbol.equals("main"))) { //$NON-NLS-1$ //$NON-NLS-2$
```
Comment (mickaelistria) [inline org.eclipse.corrosion/src/org/eclipse/corrosion/debug/RustDebugDelegate.java]: You can simply ignore the case of the .cargo project at the moment, it's not meant to be used for debug anyway

Comment (justdak) [reply org.eclipse.corrosion/src/org/eclipse/corrosion/debug/RustDebugDelegate.java]: Done

**Comment by mickaelistria** [comment]
Thank you!

**Pull request merged**